pkg/parser/node/accessor.go

```diff
@@ -43,6 +44,33 @@ func (node *Node) GetExpandedStringValue(env map[string]string) (string, error)
 	return expander.ExpandEnvironmentVariables(valueNode.Value, env), nil
 }
 
+func (node *Node) GetCredentials(env map[string]string) (string, error) {
+	switch node.Value.(type) {
+	case *ScalarValue:
+		return node.GetExpandedStringValue(env)
+	case *MapValue:
+		result := map[string]string{}
+
+		for _, child := range node.Children {
+			childValue, err := child.GetExpandedStringValue(env)
```

Comment:
I wonder if it makes sense to support nested declaration and use `GetCredentials` here as well.

One thing comes to mind here: [working with private registries](https://cirrus-ci.org/guide/linux/#working-with-private-registries):

```yaml
registry_config:
  auths:
    "container-registry.oracle.com":
      auth: ENCRYPTED[...]
```

Reply:
See https://github.com/cirruslabs/cirrus-cli/pull/538/commits/4fa0df370b3bbfd61a4200d2e7118f5e67230166.